ddtrace/propagation/http.py

```diff
@@ -130,6 +187,32 @@ def _extract(headers):
             headers,
         )
 
+        meta = None
+        tags_value = _extract_header_value(
+            _POSSIBLE_HTTP_HEADER_TAGS,
+            headers,
+            default="",
+        )
+        if tags_value:
+            # Do not fail if the tags are malformed
+            try:
+                meta = {
+                    k: v
+                    for (k, v) in decode_tagset_string(tags_value).items()
+                    if k not in _DatadogMultiHeader._X_DATADOG_TAGS_EXTRACT_REJECT
+                    if _DatadogMultiHeader._is_valid_datadog_trace_tag_key(k)
+                }
+            except TagsetMaxSizeDecodeError:
+                meta = {
+                    "_dd.propagation_error": "extract_max_size",
+                }
+                log.warning("failed to decode x-datadog-tags", exc_info=True)
```

Comment:
Would it be useful to log the size of the tag if this error ever gets thrown?

Reply:
added this to the exception message